### ISSUE TITLE
Video texture in shader not working 

### ISSUE THREAD

**Issue by brunosimon** [pixijs/pixi.js#3584]
Hello!

I updated PIXI.js on my project from 4.0.0 to 4.3.2 and everything is working but one thing, videos inside custom filters (and so shaders).

I'm sending a video texture to a shader using uniforms with `sampler2D` type.
With 4.0.0, it worked. With 4.3.2, It doesn't work anymore and the texture doesn't appears most of the time (juste some random blinks).

The base video is working. If I use a classic sprite, the video is displaying and playing without error.
The shader is also working fine. When the video is pause, it's display well. I tried with an image instead of the video, it's also working.

It's probably some settings I haven't heard of. Or maybe I need to update the uniform.

I hope it's was clear enough and my english wasn't too bad.

Thanks

**Comment by GoodBoyDigital** [comment]
Thanks for sharing! If you could provide a fiddle for us to debug that would be ace 👍

**Comment by englercj** [comment]
Hey @brunosimon have you had a chance to create a code sample we can use to debug this yet?

**Comment by brunosimon** [comment]
Hey

I tried to set a sample but a had too much issues to reproduce the exact situation for the problem. I then found a solution and moved on. Sorry, I completely forgot to answer.

It seems that the problem only occurs when we send the texture to the custom filter, but it's working fine when the filter is directly applied to the video sprite (sprite based on the video texture).
We can get the texture inside the shader with the classic `texture2D(uSampler, vTextureCoord)` without having the glitches.

I also found that sending the video to a canvas and then updating the texture based on the canvas is more efficient. I got a better framerate using this technic. 

I hope it can help.
